tensornetwork/tn_keras/conv2d_mpo.py

```diff
@@ -0,0 +1,263 @@
+    self.bias_regularizer = regularizers.get(bias_regularizer)
+
+  def build(self, input_shape: List[int]) -> None:
+    # Disable the attribute-defined-outside-init violations in this function
+    # pylint: disable=attribute-defined-outside-init
+    if self.data_format == 'channels_first':
+      channel_axis = 1
+    else:
+      channel_axis = -1
+    if input_shape[channel_axis] is None:
+      raise ValueError('The channel dimension of the inputs '
+                       'should be defined. Found `None`.')
+
+    def is_perfect_root(n, n_nodes):
+      root = n**(1. / n_nodes)
+      return round(root)**n_nodes == n
+
+    channels = input_shape[channel_axis]
+
+    # Ensure dividable dimensions
+    assert is_perfect_root(channels, self.num_nodes), \
+      f'Input dim incorrect.\
+      {input_shape[-1]}**(1. / {self.num_nodes}) must be round.'
+
+    assert is_perfect_root(self.filters, self.num_nodes), \
+      f'Output dim incorrect. \
+      {self.filters}**(1. / {self.num_nodes}) must be round.'
+
+    super(Conv2DMPO, self).build(input_shape)
+
+    in_leg_dim = math.ceil(channels**(1. / self.num_nodes))
+    out_leg_dim = math.ceil(self.filters**(1. / self.num_nodes))
+
+    self.nodes.append(
+        self.add_weight(name='end_node_first',
+                        shape=(in_leg_dim, self.kernel_size[0],
+                               self.bond_dim, out_leg_dim),
+                        trainable=True,
+                        initializer=self.kernel_initializer,
+                        regularizer=self.kernel_regularizer))
+    for i in range(self.num_nodes - 2):
+      self.nodes.append(
+          self.add_weight(name=f'middle_node_{i}',
+                          shape=(in_leg_dim, self.bond_dim, self.bond_dim,
+                                 out_leg_dim),
+                          trainable=True,
+                          initializer=self.kernel_initializer,
+                          regularizer=self.kernel_regularizer))
+    self.nodes.append(
+        self.add_weight(name='end_node_last',
+                        shape=(in_leg_dim, self.bond_dim,
+                               self.kernel_size[1], out_leg_dim),
+                        trainable=True,
+                        initializer=self.kernel_initializer,
+                        regularizer=self.kernel_regularizer))
+
+    if self.use_bias:
+      self.bias_var = self.add_weight(
+          name='bias',
+          shape=(self.filters,),
+          trainable=True,
+          initializer=self.bias_initializer,
+          regularizer=self.bias_regularizer)
+    else:
+      self.use_bias = None
+
+  def call(self, inputs: tf.Tensor, **kwargs) -> tf.Tensor:
+
+    tn_nodes = [tn.Node(n, backend='tensorflow') for n in self.nodes]
+    for i in range(len(tn_nodes) - 1):
+      tn_nodes[i][2] ^ tn_nodes[i+1][1]
+    input_edges = [n[0] for n in tn_nodes]
+    output_edges = [n[3] for n in tn_nodes]
+    edges = [tn_nodes[0][1], tn_nodes[-1][2]] + \
```

Comment:
I think you can make this just one line.

In general, we prefer to use `()` instead of `\` to break a line

So do 
```
a = (b +
     c)
```
instead of 

```
a = b + \
    c
```